### PR TITLE
Add SAML-to-OIDC federation test case

### PR DESCRIPTION
**Pull request by darshanasbg**
$subject doing both federation login and logout.

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 1]
```diff
@@ -0,0 +1,718 @@
...
+    protected OauthAdminClient adminClient;
+    private String secondaryISClientID;
+    private String secondaryISClientSecret;
+
+    private static final int PORT_OFFSET_0 = 0;
+    private static final int PORT_OFFSET_1 = 1;
+
+    CookieStore cookieStore = new BasicCookieStore();
+    private CloseableHttpClient client;
+//private static final String HTTP_RESPONSE_HEADER_LOCATION = "location";
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: remove commented code.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 2]
```diff
@@ -0,0 +1,718 @@
...
+        client = HttpClientBuilder.create().build();
+
+        boolean userCreated = addUserToSecondaryIS();
+        Assert.assertTrue(userCreated, "User creation failed in secondary IS.");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void endTest() throws Exception {
+        try {
+            super.deleteSAML2WebSSOConfiguration(PORT_OFFSET_0, PRIMARY_IS_SAML_ISSUER_NAME);
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: no need to delete saml config separately. Deleting the SP will delete this config as well.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 3]
```diff
@@ -0,0 +1,718 @@
...
+            super.deleteServiceProvider(PORT_OFFSET_0, PRIMARY_IS_SP_NAME);
+            super.deleteIdentityProvider(PORT_OFFSET_0, PRIMARY_IS_IDP_NAME);
+
+            super.deleteOIDCConfiguration(PORT_OFFSET_1, secondaryISClientID);
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: no need to delete oidc config separately. Deleting the SP will delete this config as well.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 4]
```diff
@@ -0,0 +1,715 @@
...
+
+    private void deleteAddedUsersInSecondaryIS() throws RemoteException, UserAdminUserAdminException {
+
+        UserManagementClient usrMgtClient = new UserManagementClient(getSecondaryISURI(), "admin", "admin");
+        usrMgtClient.deleteUser(SECONDARY_IS_TEST_USERNAME);
+    }
+
+    protected String getSecondaryISURI() {
+
+        return String.format("https://localhost:%s/services/", DEFAULT_PORT + PORT_OFFSET_1);
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: constant?

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java [hunk 5]
```diff
@@ -0,0 +1,715 @@
...
+        urlParameters.add(new BasicNameValuePair("user_claims_consent", "true"));
+        urlParameters.add(new BasicNameValuePair("consent_select_all", "on"));
+        urlParameters.add(new BasicNameValuePair("consent_0", "on"));
+        urlParameters.add(new BasicNameValuePair("consent", "approve"));
+
+        if (consentClaims != null) {
+            urlParameters.addAll(consentClaims);
+        }
+
+        HttpResponse response = sendPostRequestWithParameters(client, urlParameters, "https://localhost:9854/oauth2" +
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oidc/OIDCIdentityFederationTestCase.java]: constant?

**Comment by thanujalk** [comment]
@darshanasbg can we change this to a PR and merge?

**Comment by darshanasbg** [comment]
This had some issues when @piraveena checked some other work.. She will send a PR with those fixes.

**Pull request closed without merging**